ui/shared/components/panel/variants/ClinGenVciLink.jsx

```diff
@@ -34,11 +41,17 @@ class ClinGenVciLink extends React.PureComponent {
     const { hgvsc } = this.props
     const { alleleId, loading, error } = this.state
 
-    return (
-      <DataLoader contentId={hgvsc} content={alleleId || error} loading={loading} load={this.load}>
+    const clingenInfo = (
+      <div>
         <a href={CLINGEN_VCI_URL} target="_blank" rel="noreferrer">In ClinGen VCI</a>
         <br />
         {error || (alleleId && <CopyToClipboardButton text={alleleId} />)}
+      </div>
+    )
+
+    return alleleId ? clingenInfo : (
```

Comment:
Thjis works, but I think this would be clearer if the conditional existed at a different level. Theres 2 different control flows that happen here - the ID is a prop and we render the info directly, or the ID is in the state and we need to render the data loader. Rather than have this one component that has that conditional check twice to acheive that behavior, it seem like it would be neater to have a `ClingenInfo` component that just handles rendering the loaded content, a `LoadedClinGenVciLink` that handle loading the ID if necessary, and then the actual exported component that would handle the conditional:
```
const ClinGenVciLink = (
  { CAID, ...props }) => (CAID ? <ClingenInfo alleleId={CAID} /> : <LoadedClinGenVciLink {...props } />
)
```

Reply:
that's reasonable and clearer, I split up the components like you suggested